app/cluster/dynamic.go

```diff
@@ -187,13 +187,13 @@ func (d *Dynamic) handleWorkspaceChange(ctx context.Context, workspaces string)
 }
 
 func (d *Dynamic) handleModeChange(newMode servermode.Mode) error {
-	if !newMode.Valid() {
-		return fmt.Errorf("unsupported mode: %s", newMode)
-	}
 	if d.GatewayComponent {
 		d.logger.Info("Not transiting the server because this is only Gateway App")
 		return nil
 	}
+	if !newMode.Valid() {
```

Comment:
why did we move it down? Shouldn't we start with a valid mode for gateway too?

Reply:
In case of production environment, where gateway and processor runs separately, any mode change (valid/invalid) does not affect the gateway app, so if it is a gateway app, we skip any mode change. So in order to do early exit from this method, I reordered these conditions check. 
And as per discussion with @lvrach and @cisse21, it's least likely that we will ever introduce the mode change to Gateway app, and if we do then there will be more changes, ticket to which is [this](https://www.notion.so/rudderstacks/Improve-Cluster-Manager-to-Introduce-Gateway-App-Properly-5ca903e68f9b454697a1164488241485)